Aokana extraction support

Cqfuj opened the pull request:
Aokana datas are stored with another layer of encryption: bse files. They are composed of a small header followed by the file with only its first 64 bytes encrypted.
The decryption algorithm is taken from the [arc-reader](https://github.com/minirop/arc-reader) repository. I just translated it to c++ and added small organization changes.
Tell me what you think.

rr- commented on src/dec/bgi/bse_file_decoder.cc:
```diff
@@ -0,0 +1,110 @@
+        const auto data  = ((this->data[i] & 0xFF) - rand()) & 0xFF;
+
+        if (key & 1)
+            this->data[i] = data << shift | data >> (8 - shift);
```
Consider using `algo::rotl` and `algo::rotr`

rr- commented on src/dec/bgi/bse_file_decoder.cc:
```diff
@@ -0,0 +1,110 @@
+
+s32 BseDecoder::rand()
+{
+    const auto tmp = (((seed * 257 >> 8) + seed * 97) + 23) ^ -1496474763;
```
Please encode -1496474763 as a positive number in hex, like `0xDEADBEEF` (I believe it's `0xA6CD9B75` in this particular case)

rr- commented on src/dec/bgi/bse_file_decoder.cc:
```diff
@@ -0,0 +1,110 @@
+        bstr decode();
+
+        const u32 data_size = 64;
+    private:
```
Nitpick - please put a newline above `private`

rr- commented:
I like it, very nice :+1: BTW for the file formats in `GAMELIST.js` you can use `./tools/analyze` and check for "recognized as" entries.

Cqfuj commented:
Thanks, I followed your recomendations:
* add a new line above `private`
* use of `algo::rotl` and `algo::rotr` in `decode()`
* change -1496474763 to 0xA6CD9B75

I also did other few changes:
* removed some bitwise _and_ where they were not needed
* replaced bitwise _and_ by modulo operator when computing the shift and the index
* changed 32767 to 0x7FFF in `rand()` 
* used a rotation in `rand()`

rr- commented:
Merged, thanks.
